### PR TITLE
LPS-159181 Cannot add widget to the User's page when manage pages

### PR DESCRIPTION
**Pull request by pei-jung**

### CODE REVIEW

**Comment by pei-jung** [comment]
ci:test:sf

**Comment by Sharryxue** [comment]
Relevant test interrupted due to force push, will retrigger the ci:test:relevant.

**Comment by Sharryxue** [comment]
ci:test:relevant

File: portal-impl/src/com/liferay/portal/events/ServicePreAction.java [hunk 1]
```diff
@@ -993,14 +994,25 @@ private ThemeDisplay _initThemeDisplay(
...
-						if (_log.isWarnEnabled()) {
-							_log.warn(message);
-						}
+						Map<String, String[]> parameterMap = new HashMap<>(
+							originalHttpServletRequest.getParameterMap());
 
-						throw new NoSuchLayoutException(message);
+						parameterMap.remove("p_l_id");
+
+						if (parameterMap.isEmpty()) {
```
Comment (drewbrokke) [inline portal-impl/src/com/liferay/portal/events/ServicePreAction.java:1006]: I think with this logic, an attacker could just add an additional dummy query parameter to get around this.

**Review by drewbrokke — CHANGES_REQUESTED** [review]
@pei-jung I think we need to adjust this one. What do you think about this?

1. If the method is GET, disallow using the `p_l_id` parameter.
2. If the method is not GET, allow using the parameter if the user is signed in (we have the `signedIn` variable available to us here).

Thoughts?

**Comment by drewbrokke** [comment]
@pei-jung I went ahead and sent https://github.com/liferay-usm/liferay-portal/pull/671 to get CI started in case it's the direction we want to go. LMK what you think.

**Comment by drewbrokke** [comment]
Looks like Brian merged just merged #672. We'll let the tests complete anyways just in case.

**Comment by drewbrokke** [comment]
@pei-jung are we ok to close this one?

**Pull request closed without merging**